dgraph/cmd/graphql/schema/schema.go

```diff
@@ -571,6 +591,10 @@ func Stringify(schema *ast.Schema) string {
 	return sch.String()
 }
 
-func isIDField(fld *ast.FieldDefinition) bool {
-	return fld.Type.Name() == string(ID)
+func isIDField(defn *ast.Definition, fld *ast.FieldDefinition) bool {
+	return fld.Type.Name() == idTypeFor(defn)
+}
+
+func idTypeFor(defn *ast.Definition) string {
+	return "ID"
```

Comment:
Are there plans to have custom logic here based on `defn`? Can you add a comment if that's the case.